src/compiler/file_uploader/index.ts

```diff
@@ -30,7 +33,7 @@ export async function uploadFiles(
     }
 
     console.info(
-        'Finished uploading files. Waiting for file hashing to finish...'
+        'Finished uploading files. Waiting for all async processes to end...' // TODO what's happening is that the last bit of the uploadFile doesn't await the last chunk I think, so we do still need to add the onBeforeExit so we don't exit until that is done
```

Comment:
Maybe we can discuss this, did collecting all of these into an array and awaiting with Promise.all not work?

Reply:
This seemed more straight forward, there's no overhead or code complexity of having to collect them all if we can just wait for them all to be done at the end right?

Reply:
We're in the middle of accomplishing this